tests/integration/api/test_datasources_crud.py

```diff
@@ -0,0 +1,204 @@
+        assert response.status_code == HTTPStatus.CREATED, (
+            f"Setup failed: Could not create database '{unique_name}'. Response: {response.text}"
+        )
+
+        created_db_names.append(unique_name)
+        return unique_name, response.json()
+
+    # Yield the creator function to the test function
+    yield _create_and_manage_db
+
+    # --- Teardown ---
+    # This code block is executed after the test function that uses this fixture completes
+    for db_name_to_delete in created_db_names:
+        try:
+            api_client.delete(f"{api_client.base_url}/api/databases/{db_name_to_delete}")
+        except Exception as e:
+            print(f"Warning: Teardown failed to delete database '{db_name_to_delete}'. Error: {e}")
+
+
+class TestDatabasesAPI:
+    """
+    Test suite for CRUD operations on the /api/databases endpoints.
+    """
+
+    def test_create_postgres_datasource(self, api_client):
+        """
+        Tests the successful creation of a new Postgres datasource.
+        """
+        # Use a unique name to ensure this test is self-contained and repeatable
+        db_name = f"test_create_pg_{uuid.uuid4().hex[:8]}"
+        data = {
+            "database": {
+                "name": db_name,
+                "engine": "postgres",
+                "parameters": {
+                    "user": db_user,
+                    "password": db_password,
+                    "host": db_host,
+                    "port": db_port,
+                    "database": db_name,
+                    "schema": db_schema,
+                },
+            }
+        }
+        try:
+            # ACT
+            response = api_client.post(f"{api_client.base_url}/api/databases", json=data)
+
+            # ASSERT
+            assert response.status_code == HTTPStatus.CREATED, response.text
+            response_data = response.json()
+            assert response_data["name"] == db_name
+            assert response_data["engine"] == "postgres"
+        finally:
+            # CLEANUP: Ensure the created database is always deleted
+            api_client.delete(f"{api_client.base_url}/api/databases/{db_name}")
+
+    def test_create_datasource_already_exists(self, api_client, managed_db):
+        """
+        Tests creating a datasource that already exists returns a 409 Conflict.
+        """
+        # ARRANGE: Create a database that is automatically managed (created and deleted)
+        db_name, _ = managed_db("test_conflict_db", params={"user": "test"})
+
+        existing_db_payload = {"database": {"name": db_name, "engine": "postgres", "parameters": {}}}
+
+        # ACT: Attempt to create the exact same database again
+        response_conflict = api_client.post(f"{api_client.base_url}/api/databases", json=existing_db_payload)
+
+        # ASSERT
+        assert response_conflict.status_code == HTTPStatus.CONFLICT
+
+    def test_update_existing_database(self, api_client, managed_db):
```

Comment:
FYI, some test are intersected with [testing rest api in unit tests](https://github.com/mindsdb/mindsdb/blob/6df053cbcc3571d3942927c3ee42cf5bb83595d0/tests/unit/api/http/databases_test.py#L63-L62)